File: Assets/__Script/LuaBehaviour/LuaSpriteSetNotify.cs
using System;
using DragonReborn.AssetTool;
using XLua;

// ReSharper disable once CheckNamespace
namespace DragonReborn
{
	public sealed class LuaSpriteSetNotify : LuaBehaviour, ISpriteSetNotify
	{
		private Action<LuaTable, bool, string> _onSpriteManagerSetSprite;

		protected override void OnBeforeLuaAwake()
		{
			base.OnBeforeLuaAwake();
			InternalGetLuaInstance?.Get(nameof(OnSpriteManagerSetSprite), out _onSpriteManagerSetSprite);
		}

		public void OnSpriteManagerSetSprite(bool isSuccess, string spriteName)
		{
			if (null == Instance) return;
			_onSpriteManagerSetSprite?.Invoke(Instance, isSuccess, spriteName);
		}
	}
}

File: Lua/GamePlay/Utils/LuaSpriteSetNotify.lua
---@class LuaSpriteSetNotify
local LuaSpriteSetNotify = sealedClass('LuaSpriteSetNotify')

function LuaSpriteSetNotify:ctor()
    ---@type fun(isSuccess:boolean,spriteName:string)
    self._callback = nil
end

---@param isSuccess boolean
---@param spriteName string
function LuaSpriteSetNotify:OnSpriteManagerSetSprite(isSuccess,spriteName)
    if self._callback then
        self._callback(isSuccess, spriteName)
    end
end

return LuaSpriteSetNotify